src/sfizz/Wavetables.h

```diff
@@ -0,0 +1,177 @@
+   but as logarithmic divisions of the frequency range.
+ */
+class WavetableRange {
+public:
+    float minFrequency = 0;
+    float maxFrequency = 0;
+
+    static constexpr unsigned countOctaves = 10;
+    static constexpr float frequencyScaleFactor = 0.05;
+
+    static unsigned getOctaveForFrequency(float f);
+    static WavetableRange getRangeForOctave(int o);
+    static WavetableRange getRangeForFrequency(float f);
+
+    // Note: using the frequency factor 0.05, octaves are as follows:
+    //     octave 0: 20 Hz - 40 Hz
+    //     octave 1: 40 Hz - 80 Hz
+    //     octave 2: 80 Hz - 160 Hz
+    //     octave 3: 160 Hz - 320 Hz
+    //     octave 4: 320 Hz - 640 Hz
+    //     octave 5: 640 Hz - 1280 Hz
+    //     octave 6: 1280 Hz - 2560 Hz
+    //     octave 7: 2560 Hz - 5120 Hz
+    //     octave 8: 5120 Hz - 10240 Hz
+    //     octave 9: 10240 Hz - 20480 Hz
+};
+
+/**
+   Multisample of a wavetable, which is a collection of FFT-filtered mipmaps
+   adapted for various playback frequencies.
+ */
+class WavetableMulti {
+public:
+    // number of tables in the multisample
+    unsigned tableSize() const { return _tableSize; }
+
+    // number of tables in the multisample
+    static constexpr unsigned multiSize() { return WavetableRange::countOctaves; }
+
+    // get the N-th table in the multisample
+    absl::Span<const float> getTable(unsigned index) const
+    {
+        return { getTablePointer(index), _tableSize };
+    }
+
+    // get the table which is adequate for a given playback frequency
+    absl::Span<const float> getTableForFrequency(float freq) const
+    {
+        return getTable(WavetableRange::getOctaveForFrequency(freq));
+    }
+
+    // create a multisample according to a given harmonic profile
+    // the reference sample rate is the minimum value accepted by the DSP
+    // system (most defavorable wrt. aliasing)
+    static WavetableMulti createForHarmonicProfile(
+        const HarmonicProfile& hp, double amplitude, unsigned tableSize, double refSampleRate = 44100.0);
+
+    // create the tiniest wavetable with null content for use with oscillators
+    static WavetableMulti createSilence();
+
+private:
+    // get a pointer to the beginning of the N-th table
+    const float* getTablePointer(unsigned index) const
+    {
+        return &_multiData[index * (_tableSize + _tableExtra)];
+    }
+
+    // allocate the internal data for tables of the given size
+    void allocateStorage(unsigned tableSize);
+
+    // fill extra data at table ends with repetitions of the first samples
+    void fillExtra();
+
+    // length of each individual table of the multisample
+    unsigned _tableSize = 0;
+
+    // number X of extra elements, for safe interpolations up to X-th order.
+    static constexpr unsigned _tableExtra = 4;
+
+    // internal storage, having `multiSize` rows and `tableSize` columns.
+    std::unique_ptr<float[]> _multiData;
```

Comment:
Should we use the Buffer here too, just in case?

Reply:
No problem yes